Explorer/Assets/DCL/Tests/Editor/ValidationTests.cs

```diff
@@ -79,6 +81,47 @@ public void CheckForDebugUsage()
             }
         }
 
+        [Test]
+        public void CheckUnityObjectsForMissingReferences()
+        {
+            IEnumerable<ScriptableObject> scriptableObjects = GetAllScriptableObjectsInFolder("Assets/DCL");
+
+            foreach (ScriptableObject scriptableObject in scriptableObjects)
+            {
+                if (!SerializationUtility.HasManagedReferencesWithMissingTypes(scriptableObject))
+                    continue;
+
+                ManagedReferenceMissingType[] missingTypes = SerializationUtility.GetManagedReferencesWithMissingTypes(scriptableObject);
+
+                var report = new StringBuilder();
+                var missingClasses = new HashSet<string>();
+
+                foreach (ManagedReferenceMissingType missingType in missingTypes)
+                    missingClasses.Add(MissingClassFullName(missingType));
+
+                foreach (string missingClass in missingClasses)
+                    report.Append("\t").Append(missingClass).AppendLine();
+
+                Assert.Fail($"Missing references found in the following ScriptableObjects:\n{string.Join("\n", scriptableObject)}, {report}");
+            }
+        }
+
+        private static string MissingClassFullName(ManagedReferenceMissingType missingType)
+        {
+            var description = new StringBuilder();
+
+            if (missingType.namespaceName.Length > 0)
+                description.Append(missingType.namespaceName).Append(".");
+
+            description.AppendFormat("{0}, {1}", missingType.className, missingType.assemblyName);
+            return description.ToString();
+        }
+
+        private static IEnumerable<ScriptableObject> GetAllScriptableObjectsInFolder(string folderPath) =>
+            AssetDatabase.FindAssets("t:Object", new[] { folderPath })
+                         .Select(guid => AssetDatabase.LoadAssetAtPath<ScriptableObject>(AssetDatabase.GUIDToAssetPath(guid)))
+                         .ToArray();
```

Comment:
A little mention: you can return IEnumerable without turning it to array, Select() already returns IEnumerable

Reply:
I will merge it and correct in my current branch (if it is ok for you).

Reply:
good, it's not critical

Reply:
it will be here https://github.com/decentraland/unity-explorer/pull/275/commits/d0ddc4677a06bf471f8d51da1cd37d095a78be69